tests/utils/stl-lit/stl-lit.in

```diff
@@ -29,6 +29,27 @@ sys.path.insert(0, os.path.join("@LLVM_SOURCE_DIR@", 'utils', 'lit'))
 builtin_parameters= {}
 builtin_parameters['config_map'] = config_map
 
+def assert_target_arch_and_platform_equal():
+    import os
+    first_lit_cfg = next(iter(builtin_parameters['config_map'].values()))
+    target_arch = ''
+    platform = os.getenv('Platform', default='')
+    # Do nothing if "Platform" is not defined.
+    if platform != '':
+        with open(first_lit_cfg, 'r') as file:
+            for line in file:
+                if 'lit_config.target_arch =' in line:
+                    words = line.split()
+                    target_arch = words[-1].strip('\'')
+                    break
+        # arm64ec doesn't have its own command prompt with Platform.
+        if target_arch == 'arm64ec':
+            target_arch = 'arm64'
+        if platform != target_arch:
+            print('The architecture of the compiler and STL does not match')
+            exit(1)

```

Comment:
We can just pass a string to `exit()`. It would also be helpful to print how they differ.